### ISSUE TITLE
Checkbox component using wrong aria-label when it is used as a toggle

### ISSUE THREAD

**Issue by gabriel-cantergiani** [microsoft/fluentui#24612]
### Environment Information

- **Package version(s)**: @fluentui/react-northstar: 0.63.0

### Describe the issue:

Checkbox component should apply the `aria-role=switch` attribute automatically when the `toggle` prop is set to true.

#### Actual behavior:

Even when the Checkbox is configured as a toggle, the `aria-role` attribute is being set as `checkbox`, which can cause accessibility issues when using screen readers.

#### Expected behavior:

`aria-role` should be defined as `switch` if the Checkbox is configured as a toggle.

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal
